enacts/calc.py

```diff
@@ -121,6 +126,126 @@ def synthesize_enacts(variable, time_res):
     var_name = variable
     return xrds[var_name]
 
+
+def sql2GeoJSON(shapes_sql, db_sql):
+    """ Form a GeoJSON dict from sql request to a database
+
+    Parameters
+    ----------
+    shapes_sql: str
+        sql request
+    db_sql: dict
+        dictionary with host, port, user and dbname information
+    
+    Returns
+    -------
+    features: dict
+        dictionary with features as key and GeoJSON of shapes_sql as value
+
+    See Also
+    --------
+    sql2geom, geom2GeoJSON
+
+    Examples
+    --------
+    shapes_sql: select id_1 as key, name_1 as label,
+        ST_AsBinary(the_geom) as the_geom from sen_adm1
+    db_sql:
+        host: postgres
+        port: 5432
+        user: ingrid
+        dbname: iridb
+
+    Notes
+    -----
+    Description of Return is approximative as I don't know exactly what the output is
+    """
+    return geom2GeoJSON(sql2geom(shapes_sql, db_sql))
+
+
+def geom2GeoJSON(df):
+    """ Form a GeoJSON dict from a geometric object
+
+    Parameters
+    ----------
+    df: geometric object
+        shapely geometric object
+    
+    Returns
+    -------
+    features: dict
+        dictionary with features as key and GeoJSON of `geom` as value
+
+    See Also
+    --------
+    sql2geom, shapely.MultiPolygon, shapely.geometry.mapping
+
+    Notes
+    -----
+    Description of Return is approximative as I don't know exactly what the output is
+    """
+    df["the_geom"] = df["the_geom"].apply(
+        lambda x: x if isinstance(x, MultiPolygon) else MultiPolygon([x])
+    )
+    shapes = df["the_geom"].apply(shapely.geometry.mapping)
+    for i in df.index: #this adds the district layer as a label in the dict
+        shapes[i]['label'] = df['label'][i]
+    return {"features": shapes}
+
+
+def sql2geom(shapes_sql, db_sql):
+    """ Form a geometric object from sql query to a database
+
+    Parameters
+    ----------
+    shapes_sql: str
+        sql query
+    db_sql: dict
+        dictionary with host, port, user and dbname information
+    
+    Returns
+    -------
+    geometric object
+        shapely geometric object
+
+    See Also
+    --------
+    psycopg2.connect, psycopg2.sql, pandas.read_sql, shapely.wkb,
+
+    Examples
+    --------
+    shapes_sql: select id_1 as key, name_1 as label,
+        ST_AsBinary(the_geom) as the_geom from sen_adm1
```

Comment:
I'm not sure why we use `ST_AsBinary` and then use shapely to convert to GeoJSON, instead of just using [`ST_AsGeoJson`](https://postgis.net/docs/ST_AsGeoJSON.html). Possibly because some of it was copy-pasted from fbfmaproom, where shapes are rasterized into the tiles, not sent to the browser as GeoJSON. Something for me to look into.

Reply:
Even in those cases we get the result in AsBinary format (don't we?), and could choose to use AsGeoJson instead.

I'm not asking for a change right now, in any case.